provider/maas/environ.go

```diff
@@ -1322,13 +1323,25 @@ func (environ *maasEnviron) StopInstances(ids ...instance.Id) error {
 // Due to how this works in the HTTP API, an empty "ids"
 // matches all instances (not none as you might expect).
 func (environ *maasEnviron) acquiredInstances(ids []instance.Id) ([]instance.Instance, error) {
-	filter := getSystemIdValues("id", ids)
-	filter.Add("agent_name", environ.ecfg().maasAgentName())
-	return environ.instances(filter)
+	if environ.usingMAAS2() {
+		systemIDs := make([]string, len(ids))
+		for index, id := range ids {
+			systemIDs[index] = string(id)
+		}
+		args := gomaasapi.MachinesArgs{
+			AgentName: environ.ecfg().maasAgentName(),
+			SystemIDs: systemIDs,
+		}
+		return environ.instances2(args)
+	} else {
+		filter := getSystemIdValues("id", ids)
+		filter.Add("agent_name", environ.ecfg().maasAgentName())
+		return environ.instances1(filter)
```

Comment:
I'd put the maas1 up top, since it's shorter, and then drop the else, since you're returning in the first block.